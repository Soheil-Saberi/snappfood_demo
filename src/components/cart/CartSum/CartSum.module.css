.divSum{
    @apply mt-2 flex flex-row items-center justify-between
}

.divSum p{
    @apply text-sm 
}