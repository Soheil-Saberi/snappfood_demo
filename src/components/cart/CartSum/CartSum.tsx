import { FC, Fragment } from 'react';
// Style
import styles from './CartSum.module.css';

interface Props {
  sum: string;
}

const CartSum: FC<Props> = ({ sum }) => {
  return (
    <Fragment>
      {sum && (
        <div className={styles.divSum}>
          <p>
            <b>مجموع</b>
          </p>
          <p>
            <b>{sum} </b> تومان
          </p>
        </div>
      )}
    </Fragment>
  );
};

export default CartSum;
