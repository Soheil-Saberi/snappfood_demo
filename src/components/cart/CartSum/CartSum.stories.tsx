import { ComponentMeta, ComponentStory } from '@storybook/react';
import { camaPrice } from '../../../utility/Utils';
import CartSum from './CartSum';
import { string } from 'prop-types';

export default {
  component: CartSum,
  title: 'Cart/Sum'
} as ComponentMeta<typeof CartSum>;

export const CartSumStory: ComponentStory<typeof CartSum> = (args) => {
  return (
    <div className="mt-4 gap-2 p-4 flex flex-col  border rounded-md shadow-md w-full h-fit">
      <CartSum sum={camaPrice(args.sum)!} />
    </div>
  );
};

CartSumStory.args = {
  sum: '5000000'
};

CartSumStory.argTypes = {
  sum: string.isRequired
};
