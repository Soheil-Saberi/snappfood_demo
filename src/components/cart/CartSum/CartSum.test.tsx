import renderer from 'react-test-renderer';
import { describe, expect, test } from '@jest/globals';
import { cleanup } from '@testing-library/react';
// Component
import CartSum from './CartSum';

afterEach(() => {
  cleanup();
});

describe('CartRow Component', () => {
  test('Component Rendernig With Value Template', () => {
    const cartSum = renderer.create(<CartSum sum="500000" />);
    expect(cartSum.toJSON()).toMatchSnapshot();
  });

  test('Component Rendernig With Empty Value', () => {
    const cartSum = renderer.create(<CartSum sum="" />);
    expect(cartSum.toJSON()).toBeNull();
  });
});
