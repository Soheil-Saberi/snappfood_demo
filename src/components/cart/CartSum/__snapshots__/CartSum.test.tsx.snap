// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartRow Component Component Rendernig With Value Template 1`] = `
<div>
  <p>
    <b>
      مجموع
    </b>
  </p>
  <p>
    <b>
      500000
       
    </b>
     تومان
  </p>
</div>
`;
